resources/default.rb

```diff
@@ -140,6 +142,38 @@
     end
   end
 
+  nexus3_license 'update installed license' do
+    action :nothing
+    fingerprint new_resource.license_fingerprint
+    license new_resource.license
+  end
+
+  license_file_path = new_resource.properties_variables['nexus.licenseFile']
+  license_already_installed = !license_file_path.nil? && ::File.exist?(license_file_path)
+
+  directory "license directory for #{new_resource.instance_name}" do
+    path(lazy { ::File.dirname(license_file_path) })
+    recursive true
+    owner     'root'
+    group     'root'
+    mode      '0755'
+    only_if { !new_resource.license.nil? && !license_file_path.nil? }
+  end
+
+  file "license for #{new_resource.instance_name}" do
```

Comment:
Why not managing the license file in the `nexus3_license` resource?

If your concern is about the case where nexus is down, I guess it can be easily handled in the `load_current_resource`

Reply:
Tried, but did not work as I wanted it to. Especially around the notification on itself for the update.

> I guess it can be easily handled in the load_current_resource
We'd need  a custom error, and did not want to do that or rescue standard error.

Reply:
Could be interesting to discuss it a bit more in details, why would you need a notif and especially why "on itself"?

Reply:
as discussed, remove the extra notifs for restart and moved the license to the end